be/src/exec/schema_scan_node.cpp

```diff
@@ -127,7 +132,8 @@ Status SchemaScanNode::prepare(RuntimeState* state) {
         return Status::InternalError("schema scanner get nullptr pointer.");
     }
 
-    RETURN_IF_ERROR(_schema_scanner->init(&_scanner_param, _pool));
+    RETURN_IF_ERROR(
+            _schema_scanner->init(&_scanner_param, _pool, schema_table->schema_table_type()));
```

Comment:
you could add a member schem table type in SchemaScanner so that init function could use the member instead of pass in a parameter.